sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -138,8 +138,12 @@ case class RowDataSourceScanExec(
   }
 
   // Only care about `relation` and `metadata` when canonicalizing.
-  override def preCanonicalized: SparkPlan =
-    copy(rdd = null, outputPartitioning = null, metastoreTableIdentifier = None)
+  override lazy val canonicalized: SparkPlan =
+    copy(
+      output.map(QueryPlan.normalizeExprId(_, output)),
+      rdd = null,
+      outputPartitioning = null,
+      metastoreTableIdentifier = None)
```

Comment:
@cloud-fan  is it right to remove table identifiers here, could it mistakenly link two different tables with same schema together?

Reply:
`relation: BaseRelation` will take care of the equality check. Besides, the old `preCanonicalized` already set `metastoreTableIdentifier` to None.

Reply:
> `relation: BaseRelation` will take care of the equality check. Besides, the old `preCanonicalized` already set `metastoreTableIdentifier` to None.

thanks for the reply.